bundles/org.openhab.binding.millheat/src/main/java/org/openhab/binding/millheat/internal/handler/MillheatHeaterHandler.java

```diff
@@ -0,0 +1,193 @@
+                }
+            } else if (MillheatBindingConstants.CHANNEL_FAN_ACTIVE.equals(channelUID.getId())) {
+                if (command instanceof RefreshType) {
+                    updateState(channelUID, heater.isFanActive() ? OnOffType.ON : OnOffType.OFF);
+                } else if (heater.isCanChangeTemp() && heater.getRoom() == null) {
+                    updateIndependentHeaterProperties(null, null, command);
+                } else {
+                    logger.info("Heater {} cannot change temperature and is in a room", getThing().getUID());
+                }
+            } else if (MillheatBindingConstants.CHANNEL_WINDOW_STATE.equals(channelUID.getId())) {
+                if (command instanceof RefreshType) {
+                    updateState(channelUID, heater.isWindowOpen() ? OpenClosedType.OPEN : OpenClosedType.CLOSED);
+                }
+            } else if (MillheatBindingConstants.CHANNEL_INDEPENDENT.equals(channelUID.getId())) {
+                if (command instanceof RefreshType) {
+                    updateState(channelUID, heater.getRoom() == null ? OnOffType.ON : OnOffType.OFF);
+                }
+            } else if (MillheatBindingConstants.CHANNEL_CURRENT_POWER.equals(channelUID.getId())) {
+                if (command instanceof RefreshType) {
+                    if (config.power != null) {
+                        if (heater.isHeatingActive()) {
+                            updateState(channelUID, new QuantityType<>(config.power, Units.WATT));
+                        } else {
+                            updateState(channelUID, new QuantityType<>(0, Units.WATT));
+                        }
+                    } else {
+                        updateState(channelUID, UnDefType.UNDEF);
+                        logger.info(
+                                "Cannot update power for heater as the nominal power has not been configured for thing {}",
+                                getThing().getUID());
+                    }
+                }
+            } else if (MillheatBindingConstants.CHANNEL_TARGET_TEMPERATURE.equals(channelUID.getId())) {
+                if (command instanceof RefreshType) {
+                    if (heater.isCanChangeTemp() && heater.getTargetTemp() != null) {
+                        updateState(channelUID, new QuantityType<>(heater.getTargetTemp(), Units.CELSIUS));
+                    } else if (heater.getRoom() != null) {
+                        Integer targetTemperature = heater.getRoom().getTargetTemperature();
+                        if (targetTemperature != null) {
+                            updateState(channelUID, new QuantityType<>(targetTemperature, Units.CELSIUS));
+                        } else {
+                            updateState(channelUID, UnDefType.UNDEF);
+                        }
+                    } else {
+                        logger.info(
```

Comment:
Should something be marked OFFLINE, can this even happen? How would it happen?

Reply:
The API is undocumented, changes are meant to be detected easily so the binding can be updated. This logging statement is intended to do so. The semantics around independent heaters are slightly unclear.